doc/references/taxonomy-parser.md

```diff
@@ -0,0 +1,161 @@
+1. each *tag* as a *normalized version* referred to as *tagid*
+1. each *entry line* as a first tag which is referred to as *line tag*,
+   and as *line tagid* for it's *normalized version*.
+   *line tagid* is considered the identifier for this *entry line*,
+   and must be unique.
+1. the first *entry line* of each *entry block*
+   is referred to as *canonical tag* (resp. *canonical tagid*)
+   for this block.
+   This is considered the identifier for this block, and must be unique.
+1. the text in the *parent line* is composed of `LC:` followed by text.
+   This text is referred to as the *parent tag*
+
+## Sanitizing a line
+
+Each line we read in the file must be sanitized.
+
+This is the following process:
+
+1. remove any space characters at end of line
+   ```python
+   line = line.rstrip()
+   ```
+1. replace `’` (typographique quote) to simple quote `'`
+   ```python
+   line = line.replace("’", "'")
+   ```
+1. replace commas that have no space around by a lower comma character
+   and do the same for escaped comma (preceded by a `\`)
+   (to distinguish them from commas acting as tags separators)
+   ```python
+   line = re.sub(r"(\d),(\d)", r"\1‚\2", line)
+   line = re.sub(r"\\,", "\\‚", line)
+   ```
+1. removes parenthesis for roman numerals
+   ```python
+   line = re.sub(r"\(([ivx]+)\)", r"\1", line, flags=re.I)
+   ```
+
+## Normalized version of a string (of a tag)
+
+The *normalized version* of a string is the operation
+we will apply that to tag to obtain the tagid.
+
+It depends on the language.
+
+Obtain NFC form of the string. In python:
+```python
+string = unicodedata.normalize("NFC", my_input)
+```
+
+Then look at `string_normalization_for_lang` in `Config_off.pm` [^string_config]
+to know if following operations applies, based on language code:
+* lowercase
+* removing accents
+
+If lowercase applies and string is not a UUID, that is it does not match regexp `r"^[a-z\-]+\.[a-zA-Z0-9-_]{8}[a-zA-Z0-9-_]+$"`,
+then lowercase it. And also transform "." to "-".
+
+Replace every special character (ascii 0 to 27) and zero width space with "-":
+```python
+string = re.sub(r"[\u0000-\u0027\u200b]", "-", string)
+```
+
+Replace html escaped chars by "-":
+```python
+string = re.sub(r"&\w+;", "-", string)
+```
+
+Replace some weirds chars by "-" [^strech_weird_chars]:
+```
+string = re.sub(r"[\s!"#\$%&'()*+,\/:;<=>?@\[\\\]^_`{\|}~¡¢£¤¥¦§¨©ª«¬®¯°±²³´µ¶·¸¹º»¼½¾¿×ˆ˜–—‘’‚“”„†‡•…‰‹›€™\t]", "-", string)
+```
+
+If removing accents applies, use `unidecode.unidecode` to remove them.
+
+Finally, replace consecutive "-" by only one minus, and strip "-":
+```python
+string = re.sub(r"-+", "-", string)
+string = string.strip("-")
+```
+
+[^string_config]: of course you would have copied it to a local config file, or a dict in your code.
+
+[^strech_weird_chars]: (stretch goal: use [unicode category](https://www.compart.com/en/unicode/category) (through unicodedata module) to remove all unwanted category)
+
+# Processing
```

Comment:
Please add the **current logic/process** followed during *taxonomy validation* in this section. This would be helpful for implementation in the future.

Reply:
@alexgarel Regarding "validation of a taxonomy", there should be a *logic/code* in place currently. Adding that logic to this document, would be helpful for implementation.

Reply:
hum @stephanegigandet might tell me if I'm wrong, but I think we only verify we don't have same term twice in different entries (and currently we do not even enforce it, for we know some taxonomies have this problem).

I think we can leave it to the editor. But we should write a spec about it, you're right, but not at parsing time.